people-and-planet-ai/weather-forecasting/trainer/task.py

```diff
@@ -0,0 +1,312 @@
+
+    files = glob(os.path.join(data_path, "*.npz"))
+    dataset = (
+        Dataset.from_dict({"filename": files})
+        .map(
+            read_data_file,
+            num_proc=NUM_DATASET_READ_PROC,
+            remove_columns=["filename"],
+        )
+        .map(flatten, batched=True, num_proc=NUM_DATASET_PROC)
+    )
+    return dataset.train_test_split(train_size=train_test_ratio, shuffle=True)
+
+
+def augmented(dataset: Dataset) -> Dataset:
+    """Augment dataset by rotating and flipping the examples."""
+
+    def augment(values: list) -> np.ndarray:
+        transformed = [
+            np.rot90(values, 0, (1, 2)),
+            np.rot90(values, 1, (1, 2)),
+            np.rot90(values, 2, (1, 2)),
+            np.rot90(values, 3, (1, 2)),
+            np.flip(np.rot90(values, 0, (1, 2)), axis=1),
+            np.flip(np.rot90(values, 1, (1, 2)), axis=1),
+            np.flip(np.rot90(values, 2, (1, 2)), axis=1),
+            np.flip(np.rot90(values, 3, (1, 2)), axis=1),
+        ]
+        return np.concatenate(transformed)
+
+    return dataset.map(
+        lambda batch: {key: augment(values) for key, values in batch.items()},
+        batched=True,
+        num_proc=NUM_DATASET_PROC,
+    )
+
+
+def run(
+    data_path: str,
+    model_path: str,
+    epochs: int = EPOCHS,
+    batch_size: int = BATCH_SIZE,
+    train_test_ratio: float = TRAIN_TEST_RATIO,
+    from_checkpoint: bool = False,
+) -> None:
+    """Trains a new WeatherModel.
+
+    Args:
+        data_path: Directory path to read data files from.
+        model_path Directory path to write the trained model to.
+        epochs: Number of times to go through the training dataset.
+        batch_size: Number of training examples to learn from at once.
+        train_test_ratio: Ratio of examples to use for training and for testing.
+        from_checkpoint: Whether or not to resume from latest checkpoint.
+    """
+
+    print(f"data_path: {data_path}")
+    print(f"model_path: {model_path}")
+    print(f"epochs: {epochs}")
+    print(f"batch_size: {batch_size}")
+    print(f"train_test_ratio: {train_test_ratio}")
+    print("-" * 40)
+
+    dataset = read_dataset(data_path, train_test_ratio)
+    print(dataset)
+
+    model = WeatherModel.create(dataset["train"]["inputs"])
+    print(model.config)
+    print(model)
+
+    training_args = TrainingArguments(
+        output_dir=os.path.join(model_path, "checkpoints"),
+        per_device_train_batch_size=batch_size,
+        per_device_eval_batch_size=batch_size,
+        num_train_epochs=epochs,
+        logging_strategy="epoch",
+        evaluation_strategy="epoch",
+    )
+    trainer = Trainer(
+        model,
+        training_args,
+        train_dataset=augmented(dataset["train"]),
+        eval_dataset=dataset["test"],
+    )
+    trainer.train(resume_from_checkpoint=from_checkpoint)
+    trainer.save_model(model_path)
+
+
+if __name__ == "__main__":
```

Comment:
Same as previous. If you don't need the CLI set up, I recommend removing.

Reply:
Isn't a CLI is required for a trainer script in Vertex AI? Please let me know if there's a better or recommended way to launch a custom Vertex AI training job.